src/solver.rs

```diff
@@ -199,6 +200,46 @@ pub enum Dependencies<P: Package, VS: VersionSet, M: Eq + Clone + Debug + Displa
     Available(DependencyConstraints<P, VS>),
 }
 
+/// Some statistics about how much trouble the resolver has had with a package.
+pub struct PackageResolutionStatistics {
+    discovery_order: u32,
+    conflict_count: u32,
+}
+
+impl PackageResolutionStatistics {
+    fn new<P: Package>(pid: Id<P>, conflict_count: &Map<Id<P>, u32>) -> Self {
+        Self {
+            discovery_order: pid.into_raw() as u32,
```

Comment:
If you manually call alloc in some order, e.g. for dependencies before your actual package for a pre-fetching related reason, would that break this order?

Reply:
Technically it's the order alloc is called. So the actual values would certainly be affected by prefetching. However, prioritizes is only be called on  packages that are available for decision-making, so  the impact on prioritization should not be dramatically affected.